### PR TITLE
HDDS-7149. Update ratis version to 2.4.0 and thirdparty version to 1.0.2.

### PR DESCRIPTION
**Pull request by captainzmc**
## What changes were proposed in this pull request?

We should update  Ratis to 2.4.0  to fix [gRPC memleak](https://github.com/grpc/grpc-java/pull/9415). 
After we update ratis to 2.4.0, we will  start release ozone 1.3.0.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7149

## How was this patch tested?

UT had been update.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java [hunk 1]
```diff
@@ -353,8 +353,11 @@ public void notifyConfigurationChanged(long term, long index,
 
   @Override
   public void pause() {
-    getLifeCycle().transition(LifeCycle.State.PAUSING);
-    getLifeCycle().transition(LifeCycle.State.PAUSED);
+    final LifeCycle lc = getLifeCycle();
```
Comment (myskov) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java:356]: Are these changes connected to bumping ratis version?

Comment (adoroszlai) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java:356]: Yes, the tests are failing otherwise.

**Review by adoroszlai — APPROVED** [review]
Thanks @captainzmc for the patch.

**Comment by captainzmc** [comment]
Hi @szetszwo, Would you take another look？

**Review by JacksonYao287 — APPROVED** [review]
thanks @captainzmc for the work, LGTM +1! this is a important milestone for releasing ozone-1.3.0. Hope that will come soon.

**Comment by captainzmc** [comment]
Thanks @adoroszlai @szetszwo @JacksonYao287 for the review. I will cut branch-1.3 and start the release.

**Pull request merged**